openquake/smt/tests/comparison/comparison_test.py

```diff
@@ -0,0 +1,310 @@
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+"""
+Tests for execution of comparison module
+"""
+import os
+import shutil
+import unittest
+import pandas as pd
+from openquake.hazardlib import valid
+from openquake.hazardlib.imt import from_string
+from openquake.smt.comparison import compare_gmpes as comp
+from openquake.smt.comparison.utils_compare_gmpes import compute_matrix_gmpes, plot_trellis_util, plot_cluster_util, plot_sammons_util, plot_euclidean_util
+
+BASE_DATA_PATH = os.path.join(os.path.dirname(__file__), "data")
+
+# Defines the target values for each run in the inputted .toml file
+TARGET_RUN_NAMES = pd.Series(['Albania_1std','Albania_2std','Albania_3std'])
+TARGET_VS30 = pd.Series([800,800,600])
+TARGET_REGION = pd.Series([0,0,0])
+TARGET_DEPTHS = [30,30,30]
+TARGET_RMAX = pd.Series([300, 300, 300])
+TARGET_NSTD = pd.Series([1,2,3])
+TARGET_TRELLIS_MAG = [[5.0,6.0,7.0],[5.0,6.0,7.0],[5.0,6.0,7.0]]
+TARGET_MAG = {'0': ([5. , 5.1, 5.2, 5.3, 5.4, 5.5, 5.6, 5.7, 5.8, 5.9, 6. ,
+                     6.1, 6.2, 6.3, 6.4, 6.5, 6.6, 6.7, 6.8, 6.9]),
+              '1': ([5. , 5.1, 5.2, 5.3, 5.4, 5.5, 5.6, 5.7, 5.8, 5.9, 6. ,
+                     6.1, 6.2, 6.3, 6.4, 6.5, 6.6, 6.7, 6.8, 6.9]),
+              '2': ([5. , 5.1, 5.2, 5.3, 5.4, 5.5, 5.6, 5.7, 5.8, 5.9, 6. ,
+                     6.1, 6.2, 6.3, 6.4, 6.5, 6.6, 6.7, 6.8, 6.9])}
+TARGET_IMTS = {'0': {'imt_list': ['PGA', 'SA(0.1)', 'SA(0.5)', 'SA(1.0)']},
+ '1': {'imt_list': ['PGA', 'SA(0.1)', 'SA(0.5)', 'SA(1.0)']},
+ '2': {'imt_list': ['PGA', 'SA(0.1)', 'SA(0.5)', 'SA(1.0)','SA(2.0)']}}
+TARGET_GMPES = {'0': [valid.gsim('ChiouYoungs2014'),
+                valid.gsim('CampbellBozorgnia2014'), 
+                valid.gsim('BooreEtAl2014'), 
+                valid.gsim('KothaEtAl2020regional')],
+                '1': [valid.gsim('ChiouYoungs2014'),
+                valid.gsim('CampbellBozorgnia2014'),
+                valid.gsim('BooreEtAl2014'),
+                valid.gsim('KothaEtAl2020regional')],
+                '2': [valid.gsim('ChiouYoungs2014'),
+                valid.gsim('CampbellBozorgnia2014'),
+                valid.gsim('BooreEtAl2014'),
+                valid.gsim('KothaEtAl2020regional')]}
+
+class CompareGMPEsTestCase(unittest.TestCase):
+    """
+    Tests the creation of the configuration object (within compare_gmpes) and
+    the plotting functions (within utils_compare_gmpes).
+    """
+    @classmethod
+    def setUpClass(cls):
```

Comment:
In unittest you have a method called `setUp` that is executed before the tests and used to prepare the required info. See https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp

I am not sure you need the classmethod here (and you should replace cls with self) ... but maybe I do not understand your idea.